pkg/build/strategies/sti/postexecutorstep.go

```diff
@@ -482,3 +455,90 @@ func createCommandForExecutingRunScript(scriptsURL map[string]string, location s
 	}
 	return cmd
 }
+
+func downloadAndExtractFileFromContainer(docker dockerpkg.Docker, tar s2itar.Tar, sourcePath, destinationPath, containerID string) (api.FailureReason, error) {
+	glog.V(5).Infof("Downloading file %q", sourcePath)
+
+	fd, err := ioutil.TempFile(destinationPath, "s2i-runtime-artifact")
+	if err != nil {
+		res := utilstatus.NewFailureReason(
+			utilstatus.ReasonFSOperationFailed,
+			utilstatus.ReasonMessageFSOperationFailed,
+		)
+		return res, fmt.Errorf("could not create temporary file for runtime artifact: %v", err)
+	}
+	defer func() {
```

Comment:
The point of the defer is to only set it up if we successfully opened the file.  it's correct where it is.  you can't call fd.Close() if you failed to open/create the file.